### PR TITLE
New Quirk! Cyborg Lover!

### PR DESCRIPTION
**Pull request by SSensum13**
## About The Pull Request

This PR adds a new quirk for people, who want to play as silicon-friendly crew.

Basic quirk info:
- It costs 2 points.
- It has minor additions to person's mail goodies list (cable coils, basic cells, etc).
- It has a few simple mood events, when you pet a borg or being touched/hugged by borg.

## Why It's Good For The Game

I think it is nice to have a chance to play as ~~robo-creep~~ person who loves borgos.

## Changelog

:cl:
add: Added new quirk: Cyborg Lover!
/:cl:

### CODE REVIEW

File: code/datums/mood_events/generic_positive_events.dm [hunk 1]
```diff
@@ -182,12 +182,6 @@
 	mood_change = 6
 	timeout = 5 MINUTES
 
-/datum/mood_event/pet_borg
-	description = "I just love my robotic friends!"
-	mood_change = 3
-	timeout = 5 MINUTES
-	required_job = list(/datum/job/research_director, /datum/job/scientist, /datum/job/roboticist, /datum/job/geneticist)
```
Comment (Zergspower) [inline code/datums/mood_events/generic_positive_events.dm:189]: I do feel maybe roboticists should keep this by default honestly

**Comment by SomeoneYouProbablyKnow** [comment]
This is cute. Thoughts on adding them to the fan hud that the clown fan and mime fan quirks get.

**Comment by SomeoneYouProbablyKnow** [comment]
> can we also get the cyborg hater quirk specific to flipping borgs over ( I fucking hate robots )

This could be part of the illiterate quirk. But standalone quirk that rewards you for essentially bullying other players (loathsome as the borg players are) is not good.

**Comment by OmegaDarkPotato** [comment]
Personal preference question, but can you remove the mood event for petting a borg? It's not fun to have random jackasses run up to me and pet me wordlessly & I don't want the unwashed masses' hands on my divine chassis fnr

a smaller mood buff for being around one seems like it might work better idk

**Comment by CCC23** [comment]
> > can we also get the cyborg hater quirk specific to flipping borgs over ( I fucking hate robots )
> 
> This could be part of the illiterate quirk. But standalone quirk that rewards you for essentially bullying other players (loathsome as the borg players are) is not good.

Get flipped over stienkey

**Comment by SomeoneYouProbablyKnow** [comment]
> Personal preference question, but can you remove the mood event for petting a borg? It's not fun to have random jackasses run up to me and pet me wordlessly & I don't want the unwashed masses' hands on my divine chassis fnr

Yeah, okay, CCC is right.

Get flipped over stienky

**Comment by Jacquerel** [comment]
I will be honest I am not sure this is really exciting enough to exist?
Maybe if it granted a "display my fandom" pin in the same manner as the clown/mime ones which highlights you for cyborgs and the AI, and then grant the moodlet when they choose to interact with you that might at least be something but idk.

File: code/datums/quirks/positive_quirks/cyborg_lover.dm [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/datum/quirk/cyborg_lover
+	name = "Cyborg Lover"
+	desc = "You find silicon life forms fascinating! You like inspecting and touching their hulls and robo-bodies, as well you like being touched by their manipulators."
+	icon = FA_ICON_ROBOT
+	value = 2
+	mob_trait = TRAIT_CYBORG_LOVER
+	gain_text = span_notice("You are fascinated by silicon life forms.")
+	lose_text = span_danger("Cyborgs and other silicons aren't cool anymore.")
+	medical_record_text = "Patient reports being attracted by silicon life forms."
```
Comment (itseasytosee) [inline code/datums/quirks/positive_quirks/cyborg_lover.dm:9]: 🤨

File: code/datums/quirks/positive_quirks/cyborg_lover.dm [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	)
+
+/datum/quirk/cyborg_lover/add(client/client_source)
+	var/datum/atom_hud/fan = GLOB.huds[DATA_HUD_FAN]
+	fan.show_to(quirk_holder)
+
+// Quirk related mood_events
+
+/datum/mood_event/borg_touch
+	description = "Being touched by those manipulators is nice."
```
Comment (itseasytosee) [inline code/datums/quirks/positive_quirks/cyborg_lover.dm:24]: 🤨

File: code/_globalvars/traits/_traits.dm [hunk 4]
```diff
@@ -261,6 +261,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_MEDICAL_HUD" = TRAIT_MEDICAL_HUD,
 		"TRAIT_MESON_VISION" = TRAIT_MESON_VISION,
 		"TRAIT_MIME_FAN" = TRAIT_MIME_FAN,
+		"TRAIT_CYBORG_LOVER" = TRAIT_CYBORG_LOVER,
```
Comment (Mothblocks) [inline code/_globalvars/traits/_traits.dm:264]: abc order

File: code/game/objects/items/robot/items/generic.dm [hunk 5]
```diff
@@ -128,10 +132,17 @@
 				else if(user.zone_selected == BODY_ZONE_HEAD)
 					user.visible_message(span_warning("[user] bops [attacked_mob] on the head!"), \
 									span_warning("You bop [attacked_mob] on the head!"))
+					if(HAS_TRAIT(attacked_mob, TRAIT_CYBORG_LOVER))
+						attacked_mob.add_mood_event("borg_touch", /datum/mood_event/borg_touch)
```
Comment (Mothblocks) [inline code/game/objects/items/robot/items/generic.dm:136]: all of these `"borg_touch"` should be a define in this file

File: code/game/objects/items/robot/items/generic.dm [hunk 6]
```diff
@@ -128,10 +132,17 @@
...
 					user.do_attack_animation(attacked_mob, ATTACK_EFFECT_PUNCH)
 				else
-					user.visible_message(span_warning("[user] hugs [attacked_mob] in a firm bear-hug! [attacked_mob] looks uncomfortable..."), \
-							span_warning("You hug [attacked_mob] firmly to make [attacked_mob.p_them()] feel better! [attacked_mob] looks uncomfortable..."))
+					if(HAS_TRAIT(attacked_mob, TRAIT_CYBORG_LOVER))
+						user.visible_message(span_notice("[user] hugs [attacked_mob] in a firm bear-hug! [attacked_mob] looks satisfied!"), \
+								span_notice("You hug [attacked_mob] firmly to make [attacked_mob.p_them()] feel better! [attacked_mob] looks satisfied!"))
```
Comment (Mothblocks) [inline code/game/objects/items/robot/items/generic.dm:141]: Change these to be:

```dm
user.visible_message(
    "only one tab up",
    "trailing comma at the end",
)
```

**Comment by Mothblocks** [comment]
not sure if this should be a trait. feels better as signals?

**Comment by Elli-Skala** [comment]
Now to add the negative quirk: cyborg hater

File: code/game/objects/items/robot/items/generic.dm (hunk 6 above)
Comment (SSensum) [reply code/game/objects/items/robot/items/generic.dm:141]: done

**Comment by SSensum** [comment]
@Mothblocks, what about now?

**Comment by Admiral-Gigaton** [comment]
> can we also get the cyborg hater quirk specific to flipping borgs over ( I fucking hate robots )

time to RP with the illiterate and unlucky trait, as a flat earther, blaming every machine onstation for when i fall off an edge or the tram blows me up (unlucky quirk)

File: code/datums/quirks/positive_quirks/cyborg_lover.dm (hunk 2 above)
Comment (jlsnow301) [reply code/datums/quirks/positive_quirks/cyborg_lover.dm:9]: maybe you mean fascinated? attracted _by_ doesn't fit right

File: code/datums/quirks/positive_quirks/cyborg_lover.dm [hunk 7]
```diff
@@ -0,0 +1,19 @@
+/datum/quirk/cyborg_lover
+	name = "Cyborg Lover"
+	desc = "You find silicon life forms fascinating! You like inspecting and touching their hulls and robo-bodies, as well you like being touched by their manipulators."
+	icon = FA_ICON_ROBOT
+	value = 2
+	mob_trait = TRAIT_CYBORG_LOVER
+	gain_text = span_notice("You are fascinated by silicon life forms.")
+	lose_text = span_danger("Cyborgs and other silicons aren't cool anymore.")
+	medical_record_text = "Patient reports being attracted by silicon life forms."
```
Comment (Mothblocks) [inline code/datums/quirks/positive_quirks/cyborg_lover.dm:9]:
```suggestion
	medical_record_text = "Patient reports being fascinated by silicon life forms."
```

**Comment by optimumtact** [comment]
image

**Comment by SSensum** [comment]
Done replacing it with signals.

File: code/__DEFINES/dcs/signals/signals_mob/signals_mob_silicon.dm [hunk 8]
```diff
@@ -2,3 +2,10 @@
 #define COMSIG_PROCESS_BORGCHARGER_OCCUPANT "living_charge"
 ///sent from borg mobs to itself, for tools to catch an upcoming destroy() due to safe decon (rather than detonation)
 #define COMSIG_BORG_SAFE_DECONSTRUCT "borg_safe_decon"
+///called from /obj/item/borg/cyborghug/attack proc for cyborg lovers trait.
+#define COMSIG_BORG_TOUCH_MOB "borg_touch_mob"
+///called from /obj/item/borg/cyborghug/attack proc for cyborg lovers trait.
+#define COMSIG_BORG_HUG_MOB "borg_hug_mob"
+	#define COMSIG_BORG_HUG_SUCCESS 1
+///called from /mob/living/silicon/attack_hand proc for cyborg lovers trait.
+#define COMSIG_MOB_PAT_BORG "mob_pat_borg"
```
Comment (Mothblocks) [inline code/__DEFINES/dcs/signals/signals_mob/signals_mob_silicon.dm:11]: Remove "for cyborg lovers trait"--this is strictly an implementation detail.

File: code/__DEFINES/dcs/signals/signals_mob/signals_mob_silicon.dm [hunk 9]
```diff
@@ -2,3 +2,10 @@
 #define COMSIG_PROCESS_BORGCHARGER_OCCUPANT "living_charge"
 ///sent from borg mobs to itself, for tools to catch an upcoming destroy() due to safe decon (rather than detonation)
 #define COMSIG_BORG_SAFE_DECONSTRUCT "borg_safe_decon"
+///called from /obj/item/borg/cyborghug/attack proc for cyborg lovers trait.
+#define COMSIG_BORG_TOUCH_MOB "borg_touch_mob"
+///called from /obj/item/borg/cyborghug/attack proc for cyborg lovers trait.
+#define COMSIG_BORG_HUG_MOB "borg_hug_mob"
+	#define COMSIG_BORG_HUG_SUCCESS 1
```
Comment (Mothblocks) [inline code/__DEFINES/dcs/signals/signals_mob/signals_mob_silicon.dm:9]: Document waht this means.

File: code/datums/quirks/positive_quirks/cyborg_lover.dm [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+	. = ..()
+	RegisterSignal(new_holder, COMSIG_MOB_PAT_BORG, PROC_REF(pat_cyborg), override = TRUE)
+	RegisterSignal(new_holder, COMSIG_BORG_TOUCH_MOB, PROC_REF(touched_by_cyborg), override = TRUE)
+	RegisterSignal(new_holder, COMSIG_BORG_HUG_MOB, PROC_REF(hugged_by_cyborg), override = TRUE)
+
+/datum/quirk/cyborg_lover/remove_from_current_holder(quirk_transfer)
+	UnregisterSignal(quirk_holder, COMSIG_MOB_PAT_BORG)
+	UnregisterSignal(quirk_holder, COMSIG_BORG_TOUCH_MOB)
+	UnregisterSignal(quirk_holder, COMSIG_BORG_HUG_MOB)
+	. = ..()
```
Comment (Mothblocks) [inline code/datums/quirks/positive_quirks/cyborg_lover.dm:30]: `return ..()`

File: code/datums/quirks/positive_quirks/cyborg_lover.dm [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+/datum/quirk/cyborg_lover/add_to_holder(mob/living/new_holder, quirk_transfer, client/client_source)
+	. = ..()
+	RegisterSignal(new_holder, COMSIG_MOB_PAT_BORG, PROC_REF(pat_cyborg), override = TRUE)
+	RegisterSignal(new_holder, COMSIG_BORG_TOUCH_MOB, PROC_REF(touched_by_cyborg), override = TRUE)
+	RegisterSignal(new_holder, COMSIG_BORG_HUG_MOB, PROC_REF(hugged_by_cyborg), override = TRUE)
+
+/datum/quirk/cyborg_lover/remove_from_current_holder(quirk_transfer)
+	UnregisterSignal(quirk_holder, COMSIG_MOB_PAT_BORG)
+	UnregisterSignal(quirk_holder, COMSIG_BORG_TOUCH_MOB)
+	UnregisterSignal(quirk_holder, COMSIG_BORG_HUG_MOB)
```
Comment (Mothblocks) [inline code/datums/quirks/positive_quirks/cyborg_lover.dm:29]: Pass a list in instead.

File: code/game/objects/items/robot/items/generic.dm [hunk 12]
```diff
@@ -102,41 +121,60 @@
...
-									span_warning("You bop [attacked_mob] on the head!"))
+					user.visible_message(span_warning("[user] bops [attacked_mob] on the head!"),
+						span_warning("You bop [attacked_mob] on the head!"),
+					)
+					SEND_SIGNAL(attacked_mob, COMSIG_BORG_TOUCH_MOB)
 					user.do_attack_animation(attacked_mob, ATTACK_EFFECT_PUNCH)
 				else
-					user.visible_message(span_warning("[user] hugs [attacked_mob] in a firm bear-hug! [attacked_mob] looks uncomfortable..."), \
-							span_warning("You hug [attacked_mob] firmly to make [attacked_mob.p_them()] feel better! [attacked_mob] looks uncomfortable..."))
+					if(~SEND_SIGNAL(attacked_mob, COMSIG_BORG_HUG_MOB, user) & COMSIG_BORG_HUG_SUCCESS)
```
Comment (Mothblocks) [inline code/game/objects/items/robot/items/generic.dm:166]: "Borg hug success" means even less to me in this context. I'm not sure what you're doing here.

Also don't use `~` for this. If you want to invert the check, then `if (!(send_signal() & HUG_SUCCESS))`

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/__DEFINES/dcs/signals/signals_mob/signals_mob_silicon.dm (hunk 8 above)
Comment (SSensum) [reply code/__DEFINES/dcs/signals/signals_mob/signals_mob_silicon.dm:11]: done

File: code/datums/quirks/positive_quirks/cyborg_lover.dm (hunk 11 above)
Comment (SSensum) [reply code/datums/quirks/positive_quirks/cyborg_lover.dm:29]: did it

File: code/datums/quirks/positive_quirks/cyborg_lover.dm (hunk 10 above)
Comment (SSensum) [reply code/datums/quirks/positive_quirks/cyborg_lover.dm:30]: done

File: code/__DEFINES/dcs/signals/signals_mob/signals_mob_silicon.dm (hunk 9 above)
Comment (SSensum) [reply code/__DEFINES/dcs/signals/signals_mob/signals_mob_silicon.dm:9]: Done

**Comment by SSensum** [comment]
it should be better now.

**Pull request merged**